File: chess_java/src/chess/piecies/King.java
package chess.piecies;

import chess.Board;
import chess.Spot;
import chess.beans.KingIsUnderAttackResponse;

import java.util.ArrayList;
import java.util.List;

public class King extends Piece {

    private boolean castlingDone = false;

    public King(boolean white)
    {
        super(white);
    }

    public boolean isCastlingDone()
    {
        return this.castlingDone == true;
    }

    public void setCastlingDone(boolean castlingDone)
    {
        this.castlingDone = castlingDone;
    }


    @Override
    public List<Spot> getPathSpots(Board board, Spot start, Spot end) {
        return new ArrayList<>();
    }


    public KingIsUnderAttackResponse isKingUnderAttack(){
        // should check on all directions, whether it is under attack or not, with direction
        return new KingIsUnderAttackResponse(false, "Direction");
    }

}

File: chess_cpp/piecies/King.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "King.h"

King::King(bool white) : Piece(white)
{
}

bool King::isCastlingDone() {
    return this->castlingDone == true;
}

void King::setCastlingDone(bool castlingDone) {
    this->castlingDone = castlingDone;
}

std::vector<Spot> King::getPathSpots(Board* board, Spot* start, Spot* end) {
    return std::vector<Spot>();
}


KingIsUnderAttackResponse *King::isKingUnderAttack() {
    // should check on all directions, whether it is under attack or not, with direction
    return new KingIsUnderAttackResponse(false, "Direction");
}
